backend/src/services/pki-sync/azure-key-vault/azure-key-vault-pki-sync-schemas.ts

```diff
@@ -1,22 +1,53 @@
+import RE2 from "re2";
 import { z } from "zod";
 
 import { AppConnection } from "@app/services/app-connection/app-connection-enums";
 import { PkiSync } from "@app/services/pki-sync/pki-sync-enums";
 import { PkiSyncSchema } from "@app/services/pki-sync/pki-sync-schemas";
 
-import { AzureKeyVaultPkiSyncConfigSchema } from "./azure-key-vault-pki-sync-types";
+export const AzureKeyVaultPkiSyncConfigSchema = z.object({
+  vaultBaseUrl: z.string().url()
+});
+
+const AzureKeyVaultPkiSyncOptionsSchema = z.object({
+  canImportCertificates: z.boolean().default(false),
+  canRemoveCertificates: z.boolean().default(true),
+  certificateNameSchema: z
+    .string()
+    .optional()
+    .refine(
+      (schema) => {
+        if (!schema) return true;
+
+        const testName = schema
+          .replace(new RE2("\\{\\{certificateId\\}\\}", "g"), "")
+          .replace(new RE2("\\{\\{environment\\}\\}", "g"), "");
+        const azureNamePattern = new RE2("^[a-zA-Z0-9-]{1,127}$");
+
+        const forbiddenChars = "!@#$%^&*()+=[]{}|\\:;\"'<>,.?/~` _";
```

Comment:
for these patterns maybe we can use a CONSTANT instead so that we can use it in both the sync_list_option and here ?